File: bot/llm/rag-core/src/main/kotlin/ai/tock/bot/llm/rag/core/client/RagService.kt
/*
 * Copyright (C) 2017/2021 e-voyageurs technologies
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ai.tock.bot.llm.rag.core.client

import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.POST
import ai.tock.bot.llm.rag.core.client.models.RagQuery
import ai.tock.bot.llm.rag.core.client.models.RagResult


internal interface RagService {

    @POST("ask/v2")
    fun ask(@Body query: RagQuery): Call<RagResult>

}

File: llm/orchestrator-server/src/main/python/app/src/llm_orchestrator/services/rag/rag_service.py
#   Copyright (C) 2023-2024 Credit Mutuel Arkea
#
#   Licensed under the Apache License, Version 2.0 (the "License");
#   you may not use this file except in compliance with the License.
#   You may obtain a copy of the License at
#
#   http://www.apache.org/licenses/LICENSE-2.0
#
#   Unless required by applicable law or agreed to in writing, software
#   distributed under the License is distributed on an "AS IS" BASIS,
#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#   See the License for the specific language governing permissions and
#   limitations under the License.
#
"""Module for the RAG Service"""

from llm_orchestrator.routers.requests.requests import RagQuery
from llm_orchestrator.routers.responses.responses import RagResponse
from llm_orchestrator.services.langchain.rag_chain import execute_qa_chain


def rag(query: RagQuery, debug: bool) -> RagResponse:
    """Launch execution of the RAG chain"""
    return execute_qa_chain(query, debug)
